src/main/java/com/hackclub/hccore/listeners/PlayerListener.java

```diff
@@ -96,6 +96,20 @@ public void onAsyncPlayerChat(final AsyncPlayerChatEvent event) {
   public void onPlayerJoin(final PlayerJoinEvent event) {
     Player player = event.getPlayer();
     this.plugin.getDataManager().registerPlayer(player);
+    PlayerData data = this.plugin.getDataManager().getData(player);
+
+    // Check for slack link
+    if (data == null || data.getSlackId() == null) {
+      this.plugin.getLogger().info("Kicking " + player.getName() + " because they are not linked");
+      String code = this.plugin.getSlackBot().generateVerificationCode(player.getUniqueId().toString());
+      player.kick(text("You must link your Slack account to join the server!").color(RED)
+          .decorate(BOLD).appendNewline().appendNewline().append(
+              text("Please run ").color(NamedTextColor.WHITE)
+                  .append(text("/" + this.plugin.getConfig().get("settings.slack-link.base-command", "minecraft") + " link " + code).color(NamedTextColor.GOLD))
+                  .append(text(" in the #minecraft channel in the Slack to link your account.")).color(NamedTextColor.WHITE)));
```

Comment:
this needs to mention where / what slack to go to, probably directing to slack.hackclub.com

Reply:
im not 100% sure if you can have hover and clicklinks in kick msgs, i dont think you can?

Reply:
I tried and the hover and clicks didnt do anything

Reply:
Yeah ok that's fine, just write out the URL in full